src/test/ui/span/coerce-suggestions.stderr

```diff
@@ -63,7 +63,10 @@ error[E0308]: mismatched types
   --> $DIR/coerce-suggestions.rs:21:9
    |
 LL |     s = format!("foo");
-   |         ^^^^^^^^^^^^^^ expected `&mut String`, found struct `String`
+   |         ^^^^^^^^^^^^^^
+   |         |
+   |         expected `&mut String`, found struct `String`
+   |         here the type of `res` is inferred to be `String`
```

Comment:
Hm, if the "here the type of .." span overlaps the blame span of the type error, we probably shouldn't emit it.

Also `res`? Where's that coming from -- within the macro? We probably shouldn't mention macro-local variables.